src/constructs/iam/roles/roles.ts

```diff
@@ -1,11 +1,10 @@
 import { Role } from "aws-cdk-lib/aws-iam";
 import type { RoleProps } from "aws-cdk-lib/aws-iam";
-import { GuMigratableConstruct } from "../../../utils/mixin";
-import type { GuMigratingResource, GuStack } from "../../core";
+import type { GuStack } from "../../core";
 
-export interface GuRoleProps extends RoleProps, GuMigratingResource {}
+export type GuRoleProps = RoleProps;
 
-export class GuRole extends GuMigratableConstruct(Role) {
+export class GuRole extends Role {
```

Comment:
This construct is now a pure alias to the AWS one and isn't needed. I'd suggest we remove it, but not sure if it's correct to do here or in it's own PR?

Reply:
I think removing all constructs in this state in a single follow-up PR is a good way to break this change down